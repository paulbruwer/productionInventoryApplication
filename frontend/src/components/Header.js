import React from "react";
import { Container, Navbar, Nav } from "react-bootstrap";
import { Link } from "react-router-dom";

const Header = () => {

  const loginButton = () => {
    if (localStorage.getItem("jwt")){
      return (
        <Link className="header-link" to={"/"}>
              Logout
        </Link>
      )
    }
  }

  return(
  <div style={{ padding: "10px" }}>
    <div className="navbar-border">
      <Navbar className="navbar" bg="dark" data-bs-theme="dark">
        <Container>
          <Navbar.Brand>
            <Link className="header-link" to={"/"}>
              Production Inventory Application
            </Link>
          </Navbar.Brand>
          <Nav>
            <Link className="header-link" to={"/receiving"}>
              Receiving
            </Link>
            <Link className="header-link" to={"/production"}>
              Production
            </Link>
            <Link className="header-link" to={"/dispatch"}>
              Dispatch
            </Link>
            <Link className="header-link" to={"/admin"}>
              Admin
            </Link>
            {loginButton()}
          </Nav>
        </Container>
      </Navbar>
    </div>
    <style>{`
            .navbar-border{
                border: solid 1px #0063a3;
                border-radius: 10px;
            }
            .navbar{
                background-color: #0e416c !important;
                border-radius: 10px !important;
            }
            .header-link {
                color: rgb(255,255,255,0.7);
                text-decoration: none;
                padding-left: 5%;
                font-weight: 600;
            }
        `}</style>
  </div>
  )
};

export default Header;
