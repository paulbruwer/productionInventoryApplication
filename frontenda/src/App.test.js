import renderer from "react-test-renderer";
import Header from "./components/Header";
import { BrowserRouter } from "react-router-dom";


// test a snapshot of the header
test("renders correctly", () => {
  const tree = renderer.create(<BrowserRouter><Header /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});

// reset global fetch
global.fetch = jest.fn(()=>{
  Promise.resolve("abcd");
});

// unit test fetch
it("test login request", async () => {
  const result = await fetch("/receiving")
})