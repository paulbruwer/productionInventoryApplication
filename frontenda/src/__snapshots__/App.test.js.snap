// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "10px",
    }
  }
>
  <div
    className="navbar-border"
  >
    <nav
      className="navbar navbar navbar-expand navbar-light bg-dark"
      data-bs-theme="dark"
    >
      <div
        className="container"
      >
        <span
          className="navbar-brand"
        >
          <a
            className="header-link"
            href="/"
            onClick={[Function]}
          >
            Production Inventory Application
          </a>
        </span>
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="header-link"
            href="/receiving"
            onClick={[Function]}
          >
            Receiving
          </a>
          <a
            className="header-link"
            href="/production"
            onClick={[Function]}
          >
            Production
          </a>
          <a
            className="header-link"
            href="/dispatch"
            onClick={[Function]}
          >
            Dispatch
          </a>
          <a
            className="header-link"
            href="/admin"
            onClick={[Function]}
          >
            Admin
          </a>
        </div>
      </div>
    </nav>
  </div>
  <style>
    
            .navbar-border{
                border: solid 1px #0063a3;
                border-radius: 10px;
            }
            .navbar{
                background-color: #0e416c !important;
                border-radius: 10px !important;
            }
            .header-link {
                color: rgb(255,255,255,0.7);
                text-decoration: none;
                padding-left: 5%;
                font-weight: 600;
            }
        
  </style>
</div>
`;
